sdk/core/Azure.Core/tests/public/PatchModels/ChildPatchModel.Serialization.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Text.Json;
+using Azure.Core.Json;
+using Azure.Core.Serialization;
+
+namespace Azure.Core.Tests.PatchModels
+{
+    public partial class ChildPatchModel : IModelJsonSerializable<ChildPatchModel>
```

Comment:
nit: for now everything is still implementing the internal `IUtf8JsonSerializable`, not sure if you are cutting ties here?  The implications of this is if this is not the root you might break the chain since [our extension method here](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/assets/Generator.Shared/Utf8JsonWriterExtensions.cs#L71) won't know what to do.

Reply:
Yeah, not intending to cut ties, just didn't realize that was still a thing and left it off.  Let me know if you'd rather I include it in this PR, or if adding it is trivial and we can leave it out to have a simpler illustration - I'm fine either way.

Reply:
I would lean towards adding it since these models are going to serve as a template for what the generator should create so they should be as exact as possible.